source/PlayerInfo.cpp

```diff
@@ -917,18 +917,27 @@ void PlayerInfo::BuyShip(const Ship *model, const string &name, bool isGift)
 
 
 // Sell the given ship (if it belongs to the player).
-void PlayerInfo::SellShip(const Ship *selected)
+void PlayerInfo::SellShip(const Ship *selected, bool toStorage)
 {
 	for(auto it = ships.begin(); it != ships.end(); ++it)
 		if(it->get() == selected)
 		{
 			int day = date.DaysSinceEpoch();
-			int64_t cost = depreciation.Value(*selected, day);
+			int64_t cost;
+			CargoHold *storage = toStorage ? Storage(true) : nullptr;
+
+			if(storage)
+				cost = depreciation.Value(selected, day, 1);
+			else
+				cost = depreciation.Value(*selected, day);
 
 			// Record the transfer of this ship in the depreciation and stock info.
-			stockDepreciation.Buy(*selected, day, &depreciation);
+			stockDepreciation.Buy(*selected, day, &depreciation, !!storage);
```

Comment:
!!storage is the same as storage, probably not intended

Reply:
It looks like the idea is to cast the pointer (storage) to a bool.
I'd expect the compiler to do that implicitly, though, and if it can't, use `static_cast<bool>(storage)` instead. Much less confusing.